packages/carbon-graphs/src/js/controls/Bar/Bar.js

```diff
@@ -258,6 +258,22 @@ class Bar extends GraphContent {
         this.dataTarget,
         graph.config,
       );
+    } else if (graph.config.axis.y2.show) {
+      /* If data belongs to yAxis and shape associated with data id in yAxis is null then prepareLabelShapeItem method gets executed to add shape in yAxis
+       else if data belongs to y2Axis and shape associated with data id in y2Axis is null then prepareLabelShapeItem method gets executed to add shapes in y2Axes. */
+      if (this.config.yAxis === 'y' && (graph.svg.select(`.${styles.axisLabelYShapeContainer} svg[aria-describedby="${graphData.key}"]`).empty())) {
```

Comment:
can we save this query into a variable, then use it within the inner two conditionals? i'm not sure what to call it
```suggestion
      const yAxisContainer = graph.svg.select(`.${styles.axisLabelYShapeContainer} svg[aria-describedby="${graphData.key}"]`)
```

Reply:
This is one of the rule points we have, to not assign things to variable if it is only used in one place. If it is different for terra I can change it.

Reply:
oh, the style identifier is different. that tells me we could abstract this into a pure function that accepts a container param. take this with a grain of salt though, I'm still not super familiar with the project:
```js
const isYAxisContainerEmpty = (container) => graph.svg.select(`.${container} svg[aria-describedby="${graphData,key}"]`).empty();

if (this.config.yAxis === 'y' && isYAxisContainerEmpty(styles.axisLabelYShapeContainer)) { }
if (this.config.yAxis === 'y' && isYAxisContainerEmpty(styles.axisLabelY2ShapeContainer)) { }
```